themes/finna2/js/components/organisms/widgets/finna-md-editable/finna-md-editable.js

```diff
@@ -0,0 +1,265 @@
+/* global VuFind, finna, EasyMDE */
+
+/**
+ * Finna Markdown editable.
+ *
+ * @param {jQuery} element
+ * @constructor
+ */
+function FinnaMdEditable(element) {
+  this.element = element;
+  this.container = this.element.find('.finna-editable-container');
+  this.preview = true === this.container.data('preview');
+  this.emptyHtml = this.container.data('empty-html');
+  this.editor = null;
+
+  this.element.on('click.finnaEditable', { instance: this }, function onClickFinnaEditable(event) {
+    event.stopPropagation();
+    if (event.target.nodeName === 'A') {
+      // Do not open the editor when a link within the editable area was clicked.
+      return;
+    }
+    event.data.instance.openEditable();
+  });
+}
+
+FinnaMdEditable.prototype.eventOpenEditable = 'finna:openEditable';
+FinnaMdEditable.prototype.eventEditableClosed = 'finna:editableClosed';
+
+FinnaMdEditable.prototype.busyClass = 'finna-editable-busy';
+FinnaMdEditable.prototype.openClass = 'finna-editable-open';
+
+/**
+ * Returns the open state of the editable.
+ *
+ * @returns {boolean}
+ */
+FinnaMdEditable.prototype.isOpen = function isOpen() {
+  return this.element.hasClass(this.openClass);
+};
+
+/**
+ * Returns the busy state of the editable.
+ *
+ * @returns {boolean}
+ */
+FinnaMdEditable.prototype.isBusy = function isBusy() {
+  return this.element.hasClass(this.busyClass);
+};
+
+/**
+ * Conditionally sets the busy state of the editable.
+ *
+ * An opened editable can not be set busy.
+ *
+ * @param {boolean} busy Busy state to set.
+ *
+ * @returns {FinnaMdEditable}
+ */
+FinnaMdEditable.prototype.setBusy = function setBusy(busy) {
+  if (this.isOpen()) {
+    return this;
+  }
+  if (this.isBusy()) {
+    if (false === busy) {
+      this.element.removeClass(this.busyClass);
+    }
+  }
+  else if (true === busy) {
+    this.element.addClass(this.busyClass);
+  }
+  return this;
+};
+
+/**
+ * Conditionally opens the editable.
+ *
+ * A busy editable can not be opened.
+ *
+ * @returns {FinnaMdEditable}
+ */
+FinnaMdEditable.prototype.openEditable = function openEditable() {
+  if (this.isOpen() || this.isBusy()) {
+    return this;
+  }
+  var editableEvent = $.Event(this.eventOpenEditable, { editable: this });
+  $(document).trigger(editableEvent);
+  if (editableEvent.isDefaultPrevented()) {
+    return this;
+  }
+  this.element.addClass(this.openClass);
+
+  var instance = this;
```

Comment:
Voinee määrittää ekalla rivillä?
Aika monessa paikassa käytetään var _ = this; jolloin viitataan itse prototyyppiin ja merkkejä säästyy vähän. :D